scripts/deploy.sh

```diff
@@ -51,27 +55,45 @@ if [[ "$PUBLIC_URL" =~ ^file:///(.*)/archive.tar.gz ]]; then
 else
     export NIX_ON_DROID_FLAKE_URL="$PUBLIC_URL"
 fi
+export NIX_ON_DROID_BASE_URL="${PUBLIC_URL%/*}"
+
 log "NIX_ON_DROID_CHANNEL_URL=$NIX_ON_DROID_CHANNEL_URL"
 log "NIX_ON_DROID_FLAKE_URL=$NIX_ON_DROID_FLAKE_URL"
+log "NIX_ON_DROID_BASE_URL=$NIX_ON_DROID_BASE_URL"
 
-
-PROOT_HASH_FILE="modules/environment/login/default.nix"
 UPLOADS=()
 for arch in $ARCHES; do
     log "building $arch proot..."
     proot="$(nix build --no-link --print-out-paths ".#prootTermux-${arch}")"
+    proot_hash="$(nix-hash --type sha256 --sri "$proot"/bin/proot-static)"
+    attrs_file="modules/environment/login/proot-attrs/$arch.nix"
+    new_attrs_file="$(mktemp)"
+
+    cat >$new_attrs_file <<EOF
+# WARNING: This file is autogenerated by the deploy script. Any changes will be overridden
+{
+  url = "$NIX_ON_DROID_BASE_URL/bootstrap-$arch.zip";
+  hash = "$proot_hash";
+}
+EOF
 
-    if grep -q "$arch-linux = \"$proot\";" "$PROOT_HASH_FILE"; then
-        log "keeping $arch proot path in $PROOT_HASH_FILE"
-    elif grep -q "$arch-linux = \"/nix/store/" "$PROOT_HASH_FILE"; then
-        log "patching $arch proot path in $PROOT_HASH_FILE..."
-        grep "$arch-linux = \"/nix/store/" "$PROOT_HASH_FILE"
-        sed -i "s|$arch-linux = \"/nix/store/.*\";|$arch-linux = \"$proot\";|" "$PROOT_HASH_FILE"
-        log "            ->"
-        grep "$arch-linux = \"/nix/store/" "$PROOT_HASH_FILE"
+    if [[ ! -f $attrs_file ]]; then
+        log "warn: $attrs_file not present; creating"
+        mv $new_attrs_file $attrs_file
+        log "adding $attrs_file to git index"
+        git add $attrs_file
+    elif diff $attrs_file $new_attrs_file >/dev/null; then
+        log "no changes needed to $attrs_file"
     else
-        log "no $arch proot hash found in $PROOT_HASH_FILE!"
-        exit 1
+        log "updating contents of $attrs_file"
+        echo "<<<<<<"
+        cat $attrs_file
+        echo "======"
+        mv $new_attrs_file $attrs_file
+        cat $attrs_file
+        echo ">>>>>>"
+        log "adding $attrs_file to git index"
+        git add $attrs_file
```

Comment:
I don't think we should.

Reply:
I think we have to, otherwise it's not included in the flake. I agree it's less than ideal but I don't see an obvious alternative.